lib/dependencies-parser.ts

```diff
@@ -50,33 +50,49 @@ export interface PaketDependencies extends Array<DependencyGroup> {
 }
 
 function parseNuget(line: string): NugetDependency {
+  const commentRegex = new RegExp(`(${COMMENTS[0]}|${COMMENTS[1]}).+`);
+  const nameVersionOptionsRegex = new RegExp(/(\S+)\s*([^:\n]*)(:.*)?/);
+  const versionFirstOptionRegex = new RegExp(/\s(?=[^ ]*$)/);
+
+  const [, name, versionRangeAndFirstOption, restOptions] = line
+      .replace(NUGET, '')  // Remove 'nuget' string
+      .replace(commentRegex, '') // Remove comments from line end
+      .trim()
+      .match(nameVersionOptionsRegex); // Split into groups for further parsing
+
   const result: NugetDependency = {
-    name: '',
+    source: NUGET,
+    name,
+    versionRange: versionRangeAndFirstOption,
     options: {},
-    source: 'nuget',
-    versionRange: '',
   };
 
-  // Get position of first option.
-  const firstOptionMatch = /\s*[^\s]+\s*:/.exec(line);
-  let dependencyString = line;
-  // let optionsString = '';
-
-  if (firstOptionMatch) {
-    // Split line by position of first option.
-    dependencyString = line.substr(0, firstOptionMatch.index);
-    // optionsString = line.substr(firstOptionMatch.index).trim();
-  }
+  if (restOptions) {
+    // tslint:disable-next-line:prefer-const
+    let [version, firstOptionName] = versionRangeAndFirstOption.split(versionFirstOptionRegex);
```

Comment:
rename version to versionRange

Reply:
Doesn't look correct just because of the naming... can you add a comment or rearrange where `result` is defined?